### PR TITLE
Release version 0.23.1

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@RehanSD it's better to merge https://github.com/modin-project/modin/pull/6425 (fix for CI on master branch) before.

**Comment by RehanSD** [comment]
> @RehanSD it's better to merge https://github.com/modin-project/modin/pull/6425 (fix for CI on master branch) before.

Do you want to include #6425 in this patch?

File: modin/experimental/xgboost/xgboost_ray.py
Comment (mvashishtha) [inline modin/experimental/xgboost/xgboost_ray.py:1]: this shouldn't be in this PR, right?

Comment (RehanSD) [reply modin/experimental/xgboost/xgboost_ray.py:1]: I believe that this change is from #6425 which we wanted to include. I cherry picked the commit into this branch.

Comment (vnlitvinov) [reply modin/experimental/xgboost/xgboost_ray.py:1]: I think the process is we cherry-pick PRs into _release branch_ manually, then rebase _release PR_ (which only changes README) onto it.

Comment (RehanSD) [reply modin/experimental/xgboost/xgboost_ray.py:1]: Whoops, that's right! Did it correctly for the last commit we added, but not this one. I'll go ahead and re-PR!

**Pull request closed without merging**